### PR TITLE
[Term Entry] Dart Maps .remove()

### LINKED ISSUE #4374: [Term Entry] Dart Maps .remove()
### Reason/inspiration (optional)

"We would like a new term entry under Dart [Maps](https://www.codecademy.com/resources/docs/dart/concept/map) for the Dart Map `.remove()` method. 
Provide the entry in a markdown file under `docs/content/dart/concepts/map/terms/remove/remove.md`

The entry should
    - Have an introductory paragraph about what the method is and does.
    - A syntax section with a pseudo code block with an explanation of it.
    - An example showing the method in use.

Please refer to the [concept entry templates](https://github.com/Codecademy/docs/blob/main/documentation/concept-entry-template.md), [content standards](https://github.com/Codecademy/docs/blob/main/documentation/content-standards.md), and [markdown style guide](https://curriculum-documentation.codecademy.com/content-guidelines/markdown-style-guide/) when working on the PR for this issue."

### Entry Type (select all that apply)

New Term for Existing Concept Entry

### Code of Conduct

- [X] By submitting this issue, I agree to follow Codecademy Doc's [Code of Conduct](https://github.com/Codecademy/docs/blob/main/.github/CODE_OF_CONDUCT.md).

### For Maintainers

- [X] Labels added
- [x] Issue is assigned

### PR DESCRIPTION
**Pull request by dancikmad**
### Description

- Added the List.remove() method for Dart

### Issue Solved

- Closes #4374 

### Type of Change

- Adding a new entry

### Checklist

- [x] All writings are my own.
- [x] My entry follows the Codecademy Docs style guide.
- [ ] My changes generate no new warnings.
- [x] I have performed a self-review of my own writing and code.
- [x] I have checked my entry and corrected any misspellings.
- [x] I have made corresponding changes to the documentation if needed.
- [x] I have confirmed my changes are not being pushed from my forked `main` branch.
- [x] I have confirmed that I'm pushing from a new branch named after the changes I'm making.
- [x] I have linked any issues that are relevant to this PR in the `Issues Solved` section.

### CODE REVIEW

File: content/dart/concepts/map/terms/remove/remove.md [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+Tags:
+  - 'Methods'
+  - 'Dart'
+  - 'Lists'
+CatalogContent:
+  - 'learn-dart'
+  - 'paths/computer-science'
+---
+
+In Dart, the **`.remove()`** method is used to remove the first occurence of a specified value from the list. This method is useful when a particular element from a list has to be removed without knowing it's index. It returns true if the specified value is removed from the list.
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/remove/remove.md:16]: This entry is for Map concept so you need to write the content in reference to Maps. You have mentioned lists everywhere in the content, please replace it.

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @dancikmad, I noticed that you have mentioned `list.remove()` method in the content, please change it to `map.remove()`. Ping me when you are done with it.

**Comment by dancikmad** [comment]
@avdhoottt Hi, yeah I know. My mistake ... I realised this morning after I went through the doc again.
Will work on it today and replace and push the change. 
Thank you mate

**Comment by avdhoottt** [comment]
Hey @dancikmad Are you working on this PR?

**Comment by dancikmad** [comment]
@avdhoottt Yes, I have redone it, I will push the change later this evening. Sorry for the delay, been busy with a side project this week.

**Comment by avdhoottt** [comment]
Hey @dancikmad, I think you have mistakenly committed `package.json` and `yarn.lock`. Please can you delete them from this PR?

File: content/dart/concepts/map/terms/remove/remove.md [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+In Dart, the `.remove()` method removes an entry from a map using a specified key. It returns the value associated with the removed key, or null if the key doesn't exist in the map.
+
+## Syntax
+
+```pseudo
+map_name.remove(key)
+```
+
+- `key`: The key of the entry to be removed from the map.
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/remove/remove.md:24]: This line contains plagiarism, please change it.

File: content/dart/concepts/map/terms/remove/remove.md [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+void main() {
+  Map<String, ing> myMap = {'a': 1, 'b': 2, 'c': 3};
+  print('Original Map: $myMap');
+
+  // Removing an entry from the map
+  myMap.remove('c');
+
+  print('Map after removing entry with key "c": $myMap');
+
+}
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/remove/remove.md:40]:
```suggestion
void main() {
  Map<String, int> myMap = {'a': 1, 'b': 2, 'c': 3};
  print('Original Map: $myMap');
  // Removing an entry from the map
  myMap.remove('c');
  print('Map after removing entry with key "c": $myMap');
}
```
Code has some typos.

File: content/dart/concepts/map/terms/remove/remove.md [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+```
+
+Here is the output for the above code example:
+
+```shell
+Original Map: {a: 1, b: 2, c: 3}
+Map after removing entry with key "c": {a: 1, b: 2}
+```
+
+With this approach, the `.remove()` method removes the entry with the key `'c'` from the map and prints the updated map.
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/remove/remove.md:50]:
```suggestion
With this approach, the `.remove()` method removes the entry with the key `c` from the map and prints the updated map.
```

File: content/dart/concepts/map/terms/remove/remove.md [hunk 5]
```diff
@@ -0,0 +1,52 @@
...
+
+> **Note** If the specified key is not found in the map, the method returns `null`, and the map remains unchanged.
```
Comment (avdhoottt) [inline content/dart/concepts/map/terms/remove/remove.md:52]:
```suggestion
> **Note** If the specified key is not found in the map, it returns `null`, and the map remains unchanged.
```

**Review by avdhoottt — CHANGES_REQUESTED** [review]
Hey @dancikmad, please make these final changes!!

**Review by avdhoottt — APPROVED** [review]
Looks good for the second review!!

**Review by Sriparno08 — APPROVED** [review]
Looks good! 👍

**Pull request merged**